### ISSUE TITLE
More tests for `IndexedTxGraph` and `TxGraph`

### ISSUE THREAD

**Issue by evanlinjin** [bitcoindevkit/bdk#958]
**Describe the enhancement**  

Now that #926 is merged, let's look at having more fine-grained tests.

Please comment below if you wish to work on a test (to avoid duplicate work!).

### For `TxGraph`

- [ ] Test behavior of inserting `Anchor`s without the corresponding transaction existing in the graph. We should allow for this behavior.
    1. Insert various anchors for a non-existing `txid`. Ensure `Additions` are as expected.
    2. Merge the resultant `Additions` of the previous step, and attempt to recover a new `TxGraph` with them (this should succeed).
    3. Now, insert the actual transaction of `txid`. Ensure anchors still exist.
    4. Now, attempt recovery of the `TxGraph` again with the appended additions.
    - @remix75 

- [ ] Test `TxGraph::get_chain_position` and `TxGraph::list_chain_txs` behavior with unconfirmed conflicts. "Unconfirmed" can either mean that the txs do not have anchors, or that the anchors are not in the best chain. Both cases need to be tested. `TxGraph` should always pick the transaction with the latest `last_seen` value. What happens if the conflicts have the same `last_seen` value? - @rajarshimaitra

- [ ] Test `TxGraph::get_chain_position` and `TxGraph::list_chain_txs` behavior with unconfirmed transactions (txs with no anchors) that conflict with transactions that are confirmed. The unconfirmed transactions should not be givne a chain position or be listed. - @LagginTimes

**Comment by rajarshimaitra** [comment]
> Test TxGraph::get_chain_position and TxGraph::list_chain_txs behavior with unconfirmed conflicts. 

I will give this one a shot.

**Comment by remix7531** [comment]
> Test behavior of inserting Anchors without the corresponding transaction existing in the graph. We should allow for this behavior.

I will work on this

**Comment by remix7531** [comment]
> Merge the resultant Additions of the previous step, and attempt to recover a new TxGraph with them

> Now, attempt recovery of the TxGraph again with the appended additions.

What is meant by recovery in this context? We can insert things into a `TxGraph` and get information from it, but how is a `TxGraph` recovered?

**Comment by evanlinjin** [comment]
> What is meant by recovery in this context? We can insert things into a `TxGraph` and get information from it, but how is a `TxGraph` recovered?

When we mutate a `TxGraph`, `tx_graph::Additions` is returned. `tx_graph::Additions` can be `Append::append`-ed on top of one another to form an `Additions` that can be used to "recover" the `TxGraph:

```rust
let merged_additions = todo!("this is the appended additions from previous operations");

let mut graph = TxGraph::<AnchorImplementation>::default();
graph.apply_additions(merged_additions);
```

**Comment by LagginTimes** [comment]
>  Test TxGraph::get_chain_position and TxGraph::list_chain_txs behavior with unconfirmed transactions (txs with no anchors) that conflict with transactions that are confirmed. 

I will look into this one.

**Comment by rajarshimaitra** [comment]
Hey @LagginTimes I already made a sketch for it. Have you progressed quite a lot? Happy to chat and sync up on this to see if we have covered all the cases. If your PR is closer to finish happy to hand it over to you and move on to other parts.

**Comment by evanlinjin** [comment]
@rajarshimaitra No way have you started on that so quick.

**Comment by rajarshimaitra** [comment]
> @rajarshimaitra No way have you started on that so quick.

I included it in my PR as I was trying to cover all the cases with conflicts. PR soon.

**Comment by nondiremanuel** [comment]
@notmandatory shall we move this issue out of alpha.2 milestone or is it required?

**Comment by notmandatory** [comment]
Let's check with @evanlinjin and team on Tuesday. It probably still needs to be done but could go to an alpha.3 with any other testing, hardening, documentation, and examples improvements.

**Comment by notmandatory** [comment]
Replaced by #1063 will close when that one is completed.

**Comment by nondiremanuel** [comment]
I close this issue since #1063, which replaced it, was fixed by #1064
